### PR TITLE
New store_all_requests Config

### PR DESCRIPTION
**Pull request by bernardotavares**
Regarding https://github.com/rainlab/debugbar-plugin/issues/41
Open to feedback.
@LukeTowers is this what you had in mind?

### CODE REVIEW

File: config/config.php [hunk 1]
```diff
@@ -32,6 +32,24 @@
...
+     |--------------------------------------------------------------------------
+     |
+     | For security reasons the debugbar will only store requests by a
+     | backend user with the rainlab.debugbar.access_debugbar permission.
+     |
+     | If you are not concerned about security (for instance, you are in
+     | a local environment) you can set this to true to log all requests.
+     |
+     | Usefull if you want to log requests from Postman without having to worry
+     | about authentication.
```
Comment (LukeTowers) [inline config/config.php]:
```suggestion
     | about authentication. WARNING: All stored requests are publicly available without
     | any authentication, so do NOT enable this on an environment that's exposed to the
     | internet.

```

File: config/config.php [hunk 2]
```diff
@@ -32,6 +32,24 @@
...
+     |
+     */
+
+    'store_all_request' => false,
```
Comment (LukeTowers) [inline config/config.php]:
```suggestion
    'store_all_requests' => false,

```

File: middleware/InjectDebugbar.php [hunk 3]
```diff
@@ -37,7 +37,8 @@ public function handle($request, Closure $next)
 
         $user = BackendAuth::getUser();
 
-        if (!$user || !$user->hasAccess('rainlab.debugbar.access_stored_requests')) {
+        if ((!$user || !$user->hasAccess('rainlab.debugbar.access_stored_requests')) &&
+            !Config::get('rainlab.debugbar::store_all_request', false)) {
```
Comment (LukeTowers) [inline middleware/InjectDebugbar.php]:
```suggestion
            !Config::get('rainlab.debugbar::store_all_requests', false)) {
```

**Comment by LukeTowers** [comment]
Any input on this @bennothommo?

**Comment by bennothommo** [comment]
@LukeTowers @bernardotavares looks fine to me :)

**Comment by LukeTowers** [comment]
@bennothommo feel free to merge when ready and create a release

**Pull request merged**